docs/proposals/pod-security-context.md

```diff
@@ -0,0 +1,233 @@
+    // used must be specified.
+    // Optional: Default to false.
+    HostNetwork bool
+
+    // Uses the host's IPC namespace proposed in (https://github.com/kubernetes/kubernetes/pull/12470)
+    HostIPC bool
+
+    // The supplemental group ID that will own volumes in this pod and that all containers will run
+    // under as a supplemental group
+    SupplementalGroupID *int64
+
+    // The pod-level SELinux context
+    SELinuxContext *SELinuxOptions
+
+    // The base settings for container-level security attributes
+    ContainerDefaults *SecurityContext
+}
+```
+
+The `ContainerDefaults` field specifies the base security context of all containers in the pod.
+The containers' `securityContext` field is overlaid on the base security context to determine the
+effective security context for the container.
+
+### SecurityContext
+
+A previous version of this proposal dealt with an API refactor wherein a new object was introduced
+to hold the new set of container-level security attributes.  This introduced a number of complicated
+backward compatibility issues; it is far simpler to retain the existing `SecurityContext` type in
+our design.
+
+For now, the `SecurityContext` type will have only additive changes made.  In the V2 API, the
+`SELinuxContext` field will be removed from `SecurityContext` and will be exclusively pod level.
+For now, we are required to retain this field for backward compatibility purposes; these issues are
+explored in detail in the backward compatibility section.
+
+### Backward compatibility
+
+The existing V1 API should be backward compatible with the new API.  Backward compatibility is
+defined as:
+
+> 1.  Any API call (e.g. a structure POSTed to a REST endpoint) that worked before your change must
+>     work the same after your change.
+> 2.  Any API call that uses your change must not cause problems (e.g. crash or degrade behavior) when
+>     issued against servers that do not include your change.
+> 3.  It must be possible to round-trip your change (convert to different API versions and back) with
+>     no loss of information.
+
+Let's break down the fields of the `PodSecurityContext` and examine the backward compatibility rules
+for each class of field:
+
+1.  Fields that are purely additive have no backward compatibility concerns
+2.  For fields that are moving from the `PodSpec` to the `PodSecurityContext`:
+    1.  If the new field is set and the deprecated field is unset, the deprecated field is defaulted
+        to the value of the new field
+    2.  If the deprecated field is set and the new field is unset, the new field is defaulted to the
+        value of the deprecated field
+    3.  If the deprecated field and new field are both set, their values must match
+2.  For container-level base settings:
+    1.  If the field **is set** in the `ContainerDefaults` field, for each container:
+        1.  If the field is not set in the container's `SecurityContext`, the field in the
```

Comment:
please be more precise - what is a base field?  Never say "SecuirtyContext" - insteadp say pod.spec.secrityContext or container.securityContext